compiler/src/iree/compiler/Codegen/Common/VectorLayoutAnalysis.cpp

```diff
@@ -887,8 +888,9 @@ void EnforceLayout::visitRegionSuccessors(RegionBranchOpInterface branch,
     int64_t curr = 0;
     for (auto [forwardedLattice, inputLattice] :
          llvm::zip(forwardedLattices, inputLattices)) {
-      forwardedLattice->resolveWithPossibleConflict(inputLattice,
-                                                    *forwardedOperands[curr]);
+      ChangeResult changed = forwardedLattice->resolveWithPossibleConflict(
+          inputLattice, *forwardedOperands[curr]);
+      propagateIfChanged(forwardedLattice, changed);
```

Comment:
QQ: When do we update vs propagateIfChanged?

Reply:
They are the same, just at different places. update is called by transfer functions, which do not know about the fact that they are part of a dataflow analysis, so they are passed propagatedIfChanged as the update function. propagateIfChanged is the dataflow analysis API for update.

Reply:
+1 for this question. Could we unify these two functions so we just call update? (Doesn't have to be in this patch)

Reply:
Also could we unify resolve and resolveWithPossibleConflict? So from the dev API it's just a call to resolve(...) and depending on the args, we either get a conflict or not. This will also help unifying the forward and backward passes.